### PR TITLE
Solidity Template

### PR DESCRIPTION
**Pull request by sebastiantf**
Includes a bunch of changes discussed previously regarding Solidity Template repo

### CODE REVIEW

File: tasks/deploy/greeter.ts [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+import { task, types } from "hardhat/config";
 import { TaskArguments } from "hardhat/types";
 
 import { Greeter } from "../../src/types/Greeter";
 import { Greeter__factory } from "../../src/types/factories/Greeter__factory";
+import { SUBTASK_DEPLOY_VERIFY, SUBTASK_DEPLOY_WAIT_FOR_CONFIRMATIONS, TASK_DEPLOY_GREETER } from "../constants";
+
+task(TASK_DEPLOY_GREETER)
+  .addParam("greeting", "Say hello, be nice", "Hello, World", types.string)
+  .addParam("confirmations", "How many block confirmations to wait for", 0, types.int)
```
Comment (mihirgupta0900) [inline tasks/deploy/greeter.ts:10]: Why do we need this if we already have it in the verify task?

File: package.json [hunk 2]
```diff
@@ -1,10 +1,10 @@
 {
-  "name": "@paulrberg/solidity-template",
+  "name": "@coinvise/solidity-template",
   "description": "Template for writing Solidity smart contracts",
   "version": "1.0.0",
   "author": {
-    "name": "Paul Razvan Berg",
-    "url": "https://paulrberg.com"
+    "name": "Sebastian T F",
```
Comment (mihirgupta0900) [inline package.json]: Can you use "Coinvise" here and the organisation value in the url?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+name: Continuous Integration
+
+env:
+  INFURA_API_KEY: ${{ secrets.INFURA_API_KEY }}
+  MNEMONIC: ${{ secrets.MNEMONIC }}
```
Comment (mihirgupta0900) [inline .github/workflows/ci.yaml:5]: Why is this needed? For testing?

**Review by mihirgupta0900 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (sebastiantf) [reply .github/workflows/ci.yaml:5]: [hardhat.config.ts](https://github.com/coinvise/solidity-template/pull/1/files#diff-57cf5378f8cab20b02b279097ba6fad08eb53f747d81770045dda9c5f95effe3R22) uses the helper method [getEnvVar()](https://github.com/coinvise/solidity-template/pull/1/files#diff-36c97b5646983e5ed2835213316556fbb65223034c6fcb2deb035e6fa7a68610R1) which would fail early if the env vars are not set

So none of the hardhat tasks/tests would work if those env vars are not set

It makes more sense in the local environment when we're deploying or testing against a particular network. But since it's being used in the `hardhat.config.ts`, the CI would also fail early on `getEnvVar()`

File: tasks/deploy/greeter.ts (hunk 1 above)
Comment (sebastiantf) [reply tasks/deploy/greeter.ts:10]: This might be a bit of an overkill, but it handles the case of `--verify false` (in which case the confirmations in verify wont run), but you need to have some `--confirmations`

File: package.json (hunk 2 above)
Comment (sebastiantf) [reply package.json]: Done: https://github.com/coinvise/solidity-template/pull/1/commits/6d7485a60584ace4f491317ea557efd73ad647bc

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (mihirgupta0900) [reply .github/workflows/ci.yaml:5]: Can you make it not fail in the test environment? Not sure how safe it would be to store the Mnemonic

Comment (sebastiantf) [reply .github/workflows/ci.yaml:5]: The Mnemonic and infura api key doesnt actually have to be one that's being used / usable. It just have to be present and non-empty. The infura key wont ever be used. Mnemonic would be used to generate the test accounts for the Hardhat network.

As long as we're not using the CI to do actual deployments, these values doesn't mean much, and just have to be non-empty and present in the env. So any non-empty string would just work. Is that alright?

Comment (mihirgupta0900) [reply .github/workflows/ci.yaml:5]: I think it's fine for now, we can improve it further down the line

Comment (sebastiantf) [reply .github/workflows/ci.yaml:5]: Actually, setting them in hardhat.config.ts as below works, although we wont be enforcing the env vars
```typescript
// Ensure that we have all the environment variables we need.
// const infuraApiKey: string = getEnvVar("INFURA_API_KEY");
// const mnemonic: string = getEnvVar("MNEMONIC");

const infuraApiKey: string = process.env.INFURA_API_KEY || "";
const mnemonic: string = process.env.MNEMONIC || "";
```
